api/ccapi/include/model.h

```diff
@@ -144,6 +144,18 @@ class Model {
    */
   virtual int getLayer(const char *name, std::shared_ptr<Layer> *layer) = 0;
 
+  /**
+   * @brief     get input dimension of a model
+   * @retval std::vector<int *> input dimension
+   */
+  virtual std::vector<std::pair<int *, int>> getInputDimension() = 0;
```

Comment:
std::pair<int *, int> is also temporary, 

It can be `TensorDim`  or `TensorSpec` or something equivalent...

Reply:
lets keep TensorDim internally.